component/azstorage/block_blob.go

```diff
@@ -563,21 +565,17 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 		return blobList, nil, err
 	}
 
-	dereferenceTime := func(input *time.Time, defaultTime time.Time) time.Time {
-		if input == nil {
-			return defaultTime
-		} else {
-			return *input
-		}
-	}
-
 	// Process the blobs returned in this result segment (if the segment is empty, the loop body won't execute)
 	// Since block blob does not support acls, we set mode to 0 and FlagModeDefault to true so the fuse layer can return the default permission.
 
 	// For some directories 0 byte meta file may not exists so just create a map to figure out such directories
 	var dirList = make(map[string]bool)
 
 	for _, blobInfo := range listBlob.Segment.BlobItems {
+		if blobInfo.Name == nil {
```

Comment:
These checks were removed from Sourav's code so you might need to resync with him latest branch.